File: backend/budget-helper/src/main/java/dna/budget/helper/project/budgethelper/services/BudgetService.java
package dna.budget.helper.project.budgethelper.services;
import dna.budget.helper.project.budgethelper.dto.BudgetDto;
import java.util.List;

public interface BudgetService {
    List<BudgetDto> getBudgets();
    BudgetDto updateBudget(BudgetDto budgetDto) throws Exception;
    BudgetDto getBudget(String yearMonth) throws Exception;
    BudgetDto addDefaultBudget(BudgetDto budgetDto) throws Exception;
}

File: frontend/budgethelper-frontend/src/services/BudgetService.js
import axios from "axios";

const REST_API_BASE_URL = 'http://localhost:8080/budget-helper/api/budget';

export const getBudgets = () => axios.get(REST_API_BASE_URL);
export const addDefaultBudget = (newBudget) => axios.post(REST_API_BASE_URL + '/add-default', newBudget);
export const updateBudget = (newBudget) => axios.put(REST_API_BASE_URL + '/update-budget', newBudget);
export const getBudget = (yearMonth) => axios.get(REST_API_BASE_URL + '/' + yearMonth, yearMonth);
